### DISCUSSION TITLE
Component discovery for Java properties

### DISCUSSION THREAD

**Discussion by eheck-is** [WeblateOrg/weblate#12409] [Q&A] [answered]
I am trying to set up the Discovery extension for a Java project, but I can't figure out how to define the regular expression. Let's say I have the file structure from the docs: 
`(?P<originalHierarchy>.+/)(?P<component>[^/]*)/src/main/resources/ApplicationResources_(?P<language>[^/.]*)\.properties`
IIUC, the monolingual base file name with this would be `{{ originalHierarchy }} {{ component }}/src/main/resources/ApplicationResources.properties`, because in Java you usually define a default file that doesn't have the language code in its name.
Now assume, I create an new default file `/myParent/myComponent/src/main/resources/ApplicationResources.properties` and want to use Weblate to create translation files. Discovery doesn't detect the new file because it doesn't have the underscore and the language code and doesn't match the regex. Can Discovery be configured in a way that supports this approach? (As a workaround I could create empty translation files for each new resource, but can it be avoided?)

**Comment by nijel** [comment, marked as answer]
Presently translation file is necessary for discovery, see https://github.com/WeblateOrg/weblate/issues/2084

**Comment by eheck-is** [comment]
Thank you
